
From: Joshua Kwan <joshk@triplehelix.org>

Debian's initial ramdisk creation scripts rely on the content of /proc/scsi
to determine which module is needed for the ramdisk.  Unfortunately, bad
things happen when a bazillion different drivers use the _exact same name_
for /proc, namely the esp family.

Give them all unique names.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/scsi/dec_esp.c   |    2 +-
 25-akpm/drivers/scsi/jazz_esp.c  |    2 +-
 25-akpm/drivers/scsi/mac_esp.c   |    2 +-
 25-akpm/drivers/scsi/mca_53c9x.c |    2 +-
 25-akpm/drivers/scsi/sun3x_esp.c |    2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff -puN drivers/scsi/dec_esp.c~disambiguate-espc-clones drivers/scsi/dec_esp.c
--- 25/drivers/scsi/dec_esp.c~disambiguate-espc-clones	2004-08-27 03:34:15.256927088 -0700
+++ 25-akpm/drivers/scsi/dec_esp.c	2004-08-27 03:34:15.266925568 -0700
@@ -120,7 +120,7 @@ static int dec_esp_release(struct Scsi_H
 }
 
 static Scsi_Host_Template driver_template = {
-	.proc_name		= "esp",
+	.proc_name		= "dec_esp",
 	.proc_info		= &esp_proc_info,
 	.name			= "NCR53C94",
 	.detect			= dec_esp_detect,
diff -puN drivers/scsi/jazz_esp.c~disambiguate-espc-clones drivers/scsi/jazz_esp.c
--- 25/drivers/scsi/jazz_esp.c~disambiguate-espc-clones	2004-08-27 03:34:15.257926936 -0700
+++ 25-akpm/drivers/scsi/jazz_esp.c	2004-08-27 03:34:15.265925720 -0700
@@ -286,7 +286,7 @@ static void dma_led_on(struct NCR_ESP *e
 }
 
 static Scsi_Host_Template driver_template = {
-	.proc_name		= "esp",
+	.proc_name		= "jazz_esp",
 	.proc_info		= &esp_proc_info,
 	.name			= "ESP 100/100a/200",
 	.detect			= jazz_esp_detect,
diff -puN drivers/scsi/mac_esp.c~disambiguate-espc-clones drivers/scsi/mac_esp.c
--- 25/drivers/scsi/mac_esp.c~disambiguate-espc-clones	2004-08-27 03:34:15.259926632 -0700
+++ 25-akpm/drivers/scsi/mac_esp.c	2004-08-27 03:34:15.266925568 -0700
@@ -731,7 +731,7 @@ static void dma_setup_quick(struct NCR_E
 }
 
 static Scsi_Host_Template driver_template = {
-	.proc_name		= "esp",
+	.proc_name		= "mac_esp",
 	.name			= "Mac 53C9x SCSI",
 	.detect			= mac_esp_detect,
 	.slave_alloc		= esp_slave_alloc,
diff -puN drivers/scsi/mca_53c9x.c~disambiguate-espc-clones drivers/scsi/mca_53c9x.c
--- 25/drivers/scsi/mca_53c9x.c~disambiguate-espc-clones	2004-08-27 03:34:15.260926480 -0700
+++ 25-akpm/drivers/scsi/mca_53c9x.c	2004-08-27 03:34:15.265925720 -0700
@@ -445,7 +445,7 @@ static void dma_led_off(struct NCR_ESP *
 }
 
 static Scsi_Host_Template driver_template = {
-	.proc_name		= "esp",
+	.proc_name		= "mca_53c9x",
 	.name			= "NCR 53c9x SCSI",
 	.detect			= mca_esp_detect,
 	.slave_alloc		= esp_slave_alloc,
diff -puN drivers/scsi/sun3x_esp.c~disambiguate-espc-clones drivers/scsi/sun3x_esp.c
--- 25/drivers/scsi/sun3x_esp.c~disambiguate-espc-clones	2004-08-27 03:34:15.262926176 -0700
+++ 25-akpm/drivers/scsi/sun3x_esp.c	2004-08-27 03:34:15.267925416 -0700
@@ -370,7 +370,7 @@ static int sun3x_esp_release(struct Scsi
 }
 
 static Scsi_Host_Template driver_template = {
-	.proc_name		= "esp",
+	.proc_name		= "sun3x_esp",
 	.proc_info		= &esp_proc_info,
 	.name			= "Sun ESP 100/100a/200",
 	.detect			= sun3x_esp_detect,
_
